src/websocket-client.ts

```diff
@@ -510,7 +510,8 @@ export class WebsocketClient extends EventEmitter {
   private connectToWsUrl(url: string, wsKey: WsKey): WebSocket {
     this.logger.silly(`Opening WS connection to URL: ${url}`, { ...loggerCategory, wsKey })
 
-    const ws = new WebSocket(url);
+    const agent = this.options.requestOptions?.agent;
+    const ws = new WebSocket(url, undefined, { agent });
```

Comment:
Looks good - could you check if there's any chance in behaviour if "agent" happens to be undefined from that optional chain?

Reply:
It did work from our local testing.  I went ahead with rebasing against the latest and added a trinary to keep that original value as-is.  Let me know,

Reply:
Thanks - just wanted to make sure. It'll be available in a few minutes on the latest version, v2.1.5